app/services/submit_reference.rb

```diff
@@ -18,11 +18,39 @@ def save!
     if @send_emails
       CandidateMailer.reference_received(reference).deliver_later
       RefereeMailer.reference_confirmation_email(application_form, reference).deliver_later
+      notify_provider_users
     end
   end
 
 private
 
+  def notify_provider_users
+    application_choices.each do |ac|
+      ac.provider.provider_users.each do |pu|
```

Comment:
how about shifting this logic out into a query class - something like `GetProvidersToNotifyOnReferenceReceived` would allow us to be more explicit about what's going on.

Reply:
Yes, I think this is a good idea! :) It is marked as resolved but I can't find the new class. Did we change the direction?